src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -56,55 +55,48 @@ public EmailProcessor() {
-     * Iterate through all Email objects and triggers the webhook for each pending email object.
+     * Iterate through all Email objects and triggers the web hook for each pending email object.
      *
      * @param tenantConfig Configuration of a tenant
      * @return Statics of the sent emails
      */
     public CompletionStage<Statistics> processEmails(final TenantConfiguration tenantConfig) {
-        SphereClient client = tenantConfig.getSphereClient();
-        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID).withLimit(100L)
-            .withSort(s -> s.createdAt().sort().asc());
-        return client
-            .execute(query)
-            .thenApply(response -> {
-                Statistics statistics = new Statistics(tenantConfig.getProjectKey());
-                if (response.getTotal() < 1) {
-                    LOG.info(String.format("No email to process for tenant %s", tenantConfig
-                        .getProjectKey()));
-                }
-                List<CompletableFuture<Void>> allTenants = response.getResults().stream()
-                    .map(customObject -> Optional.ofNullable(customObject)
-                        .map(CustomObject::getValue)
-                        .map(node -> node.get(EMAIL_PROPERTY_STATUS))
-                        .map(JsonNode::asText)
-                        .filter(status -> ((equalsIgnoreCase(status, STATUS_PENDING) || tenantConfig.isProcessAll())))
-                        .map(pending -> callApiEndpoint(customObject.getId(), tenantConfig)
-                            .thenAccept(statistics::update))
-                        .orElseGet(() -> {
-                            statistics.update(Statistics.RESPONSE_IGNORED);
-                            return CompletableFuture.completedFuture(null);
-                        }))
-                    .collect(toList());
-
-                // join all the email processors
-                allOf(allTenants.toArray(new CompletableFuture[0])).join();
-
-                client.close();
-                return statistics;
-            })
-            .exceptionally(exception -> {
-                LOG.error(String.format("[%s] An unknown error occurred", tenantConfig.getProjectKey()),
-                    exception);
-                client.close();
-                return Statistics.ofError(tenantConfig.getProjectKey());
-            });
+        final SphereClient client = tenantConfig.getSphereClient();
+        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID);
+        if (!tenantConfig.isProcessAll()) {
+            query = query.plusPredicates(QueryPredicate.of("value(status=\"" + STATUS_PENDING + "\")"));
+        }
+        // We sort the email object by creation time, to ensure that the emails are delivered in the correct
+        // chronological order they were sent with.
+        query = query.withSort(s -> s.createdAt().sort().asc());
```

Comment:
@heshamMassoud @ahalberkamp We should use here pagination by UUID and not custom sort and offset (due to know limitations). If it is not supported by CTP JVM SDK yet than we need to create an issue here and address it once non offset based pagination is possible over SDK.